pkg/utils/common.go

```diff
@@ -510,6 +510,17 @@ func IsUki() bool {
 	return false
 }
 
+// IsUkiWithFs checks if the system is running in UKI mode
+// by checking the kernel command line for the rd.immucore.uki flag
+// Uses a v1.Fs interface to allow for testing
+func IsUkiWithFs(fs v1.FS) bool {
```

Comment:
this was needed in order to be able to abstract this check and be able to test it and actions that use it